File: 0884-uncommon-words-from-two-sentences/0884-uncommon-words-from-two-sentences.java
class Solution {
    public String[] uncommonFromSentences(String s1, String s2) {
        Map<String, Integer> map = new HashMap<>();
        for(String word: s1.split(" ")){
            map.put(word,map.getOrDefault(word,0)+1);
        }
        for(String word: s2.split(" ")){
            map.put(word,map.getOrDefault(word,0)+1);
        }
        List<String > res = new ArrayList<>();
        for(Map.Entry<String,Integer>entry : map.entrySet()){
            if(entry.getValue()==1){
                res.add(entry.getKey());
            }
        }
        return res.toArray(new String[0]);
        
    }
}

File: 0884-uncommon-words-from-two-sentences/0884-uncommon-words-from-two-sentences.py
class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        map={}
        for word in s1.split():
            if word in map:
                map[word]+=1
            else:
                map[word]=1
                
        for word in s2.split():
            if word in map:
                map[word]+=1
            else:
                map[word]=1
        res=[]
        for word in map:
            if map[word]==1:
                res.append(word)
        return res
        
        
        